Yaroslav Kernitskyi/task3/src/FileAnalysis.cpp

```diff
@@ -0,0 +1,177 @@
+#include "FileAnalysis.hpp"
+
+#include <cctype>
+#include <iostream>
+
+/*
+At the moment lines are classified according to following rules:
+1. only spaces, tabs or newline character -> blank_lines++
+2. contains code -> code_lines++
+3. contains comment -> comment_lines++
+4. same as 1 but inside comment -> blank_lines++, comment_lines++;
+
+code_lines + blank_lines + comment_lines != total_lines
+*/
+
+FileAnalysisInfo analyze(std::istream &file)
```

Comment:
This function is too complicated. It is better to split off switch-case to separate methods or objects something like analyzers